sympy/combinatorics/free_groups.py

```diff
@@ -1,18 +1,16 @@
 # -*- coding: utf-8 -*-
 from __future__ import print_function, division
 
-from sympy.core.basic import Basic
+from sympy.core.backend import Basic, Symbol, S
 from sympy.core.compatibility import is_sequence, as_int, string_types
 from sympy.core.expr import Expr
-from sympy.core.symbol import Symbol, symbols as _symbols
+from sympy.core.symbol import symbols as _symbols
```

Comment:
Is there a way around this? Currently importing this from `SymEngine` leads to a huge number of exceptions thrown across files, and it would take me some more time clearing this out.